.github/workflows/typescript.yml

```diff
@@ -143,3 +143,61 @@ jobs:
           npx concurrently -k -s first -n "SB,TEST" -c "magenta,blue" \
             "npx http-server storybook-static --port $PORT -a $HOST" \
             "npx wait-on http://$HOST:$PORT/ && npm run storybook:test -- --ci --url http://$HOST:$PORT/"
+
+  smoke_tests_update:
+    # Run on all events defined above, except pushes which are not to master
+    if: github.event_name != 'push' || github.ref == 'refs/heads/master'
+    runs-on: ubuntu-latest
+    name: "Storybook Smoke Tests Update"
+
+    container:
+      image: mcr.microsoft.com/playwright:v1.39.0-jammy
+      options: --user 1001
+
+    steps:
+      - name: 📖 Checkout commit locally
+        uses: actions/checkout@v3
+
+      - name: 📦 Install build dependencies
+        run: |
+          npm ci CYPRESS_INSTALL_BINARY=0 --ignore-scripts
+
+      - name: 🏗️ Build packages
+        run: |
+          npx nx run-many -t build
+
+      - name: 📚 Build Storybook
+        run: npm run build-storybook
+
+      - name: Run smoke tests
+        env:
+          PORT: 8080
+          HOST: 127.0.0.1
+
+        run: |
+          npx concurrently -k -s first -n "SB,TEST" -c "magenta,blue" \
+            "npx http-server storybook-static --port $PORT -a $HOST" \
+            "npx wait-on http://$HOST:$PORT/ && npm run storybook:test -- --updateSnapshot --ci --url http://$HOST:$PORT/"
+
+
+      # upload the failure screenshots for reference
+      - name: upload artifacts
+        uses: actions/upload-artifact@v2
+        if: github.event_name != 'release' && failure()
```

Comment:
We can remove the `failure()` condition and proceed to the next step.

We can then make the next step fail with the [`if-no-files-found` option](https://github.com/actions/upload-artifact?tab=readme-ov-file#inputs)